### PR TITLE
get disks using smartctl instead of lshw

### LINKED ISSUE #2: Get physical disks instead of virtual disks from raid controllers
Our collector is showing virtual disks like:

> 008817b21bd7d8881b00609cf760f681	223 GB	PERC H710P

Where we actually care more about the physical disks backing it.

We need to be able to recognize and dereference at least:

* Dell raid controllers
* megaraid raid controllers

Typically, we'll be running into LSI-based controllers.

There is probably some intelligent way for us to figure out whether a disk is physical or virtual.

### PR DESCRIPTION
**Pull request by dwradcliffe**
Fixes #2

@dalehamel @sroysen

### CODE REVIEW

File: lib/genesis_collector/disks.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
+module GenesisCollector
+  module Disks
+
+    def collect_disks
+      @payload[:disks] = enumerate_disks
+      @payload[:disks].each do |d|
+        info = get_disk_info(d.delete(:smartctl_cmd))
+        d[:vendor] = info.match(/^Vendor:\s+(.*)$/)[1] rescue nil
+        d[:vendor] ||= info.match(/^Device Model:\s+(Crucial)_(.*)$/)[1] rescue nil
```
Comment (dalehamel) [inline lib/genesis_collector/disks.rb]: Is this special case needed only for crucials?

**Comment by dalehamel** [comment]
@dwradcliffe sweet, so this works on nodes with LSI raid cards now?

**Comment by dalehamel** [comment]
cc @udnay

File: lib/genesis_collector/disks.rb (hunk 1 above)
Comment (dwradcliffe) [inline lib/genesis_collector/disks.rb]: Fixed to be more generic.

**Comment by dwradcliffe** [comment]
Yep seems to work with the raid cards I tested.

**Pull request merged**